src/server/nonClassEvent/__tests__/nonClassEvent.controller.test.ts

```diff
@@ -99,4 +131,44 @@ describe('NonClassEvent controller', function () {
       );
     });
   });
+  describe('create', function () {
+    it('creates non class parents within an existing area', async function () {
+      const mockArea = rawAreaList[0];
+      mockAreaRepository.findOneOrFail.resolves(mockArea);
+
+      await controller.create(createNonClassParent);
```

Comment:
I was testing this locally with `throw e` added to the catch block of `createWithNonClassEvents` and this test started failing with an error about not being able to destructure `id` -- I assume that's happening because `mockNonClassEventService.createWithNonClassEvents` is just a `stub()`, so it's not actually returning any value to destructure.

Reply:
Ah, yes - that probably also comes back to what you said about the catch block swallowing other errors.

Fixed in 98aec625686fb3acfb30afb7cd073ada0590dbcc